### PR TITLE
Snapshot strategies: newline for subquery

### PR DESCRIPTION
**Pull request by RobbertDM**
resolves https://github.com/dbt-labs/dbt-core/issues/6781

### Description
If you don't add these, the `) subq` will be appended to the last line of the snapshot query, which could be a comment line.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by RobbertDM** [comment]
CLA: signed again with case sensitive GitHub username this time :thinking:

**Comment by dbeatty10** [comment]
Thanks for opening this PR @RobbertDM !

A bookkeeping piece (which will resolve the other failing CI check you are seeing):
- [ ] Could you run `changie new` as described [here](https://github.com/dbt-labs/dbt-core/blob/d2f3cdd6de4aad85aa9d7523eb1179f03a2f1666/CONTRIBUTING.md#adding-or-modifying-a-changelog-entry) so we can generate a changelog entry and credit you?

Speaking of CI, we'll want to:
- [ ] add a test similar to [this](https://github.com/dbt-labs/dbt-bigquery/pull/458/files#diff-f85555042fc26c3922a3f2aa32dae5d2ebbb6e0651f245821363d5811e196152R29) so that we can catch any regressions in the future. Ideally, there would be a pre-existing test model to which a comment can be appended as a postfix.

**Comment by VersusFacit** [comment]
Hey @RobbertDM, thanks for your ping. I think Doug's right to make sure we have a test for this case. We have snapshot tests in [`tests/functional/simple_snapshot`](https://github.com/dbt-labs/dbt-core/tree/main/tests/functional/simple_snapshot). Basically, add to our snapshot tests a new or existing file that describes the case where the model has your (pre-merge) failing use case. Should be as simple as writing a SQL definition and copying the format for a test class.

**Comment by RobbertDM** [comment]
Thanks for the guidance. I added a test file but I have little idea of what I am doing, so please review carefully :-)

**Comment by VersusFacit** [comment]
Awesome!!  Thanks for the fast turnaround. It's basically all but there. You'll have to follow flake's directions of nixing unneeded names and such. It's perfectly fine you went with the conventions you saw. I think they pretty much look good for this case. We'll see with the adapter test results on your CI here.

**Comment by dbeatty10** [comment]
Thanks again for working on this PR @RobbertDM !

### TL;DR

I think you'll ultimately need to add lines like the following:

```py
    seeds__seed_csv,
```

and:

```py
    @pytest.fixture(scope="class")
    def seeds(self):
        return {"seed.csv": seeds__seed_csv}
```

Do you want to give that a try?

Please let us know if you run into any problems and we'll try to help out.

### More detail

I took a peek at the error during CI:
```
18:37:17  Completed with 1 error and 0 warnings:
18:37:17  
18:37:17  Database Error in snapshot snapshot_actual (snapshots/snapshot.sql)
18:37:17    relation "test16758814328907864877_test_comment_ending_snapshot.seed" does not exist
18:37:17    LINE 21:     select * from dbt.test16758814328907864877_test_comment_...
```

It's trying to snapshot a table named `dbt.test16758814328907864877_test_comment_ending_snapshot.seed`, but it's not finding that table. So I think the main thing to do is make sure it exists.

I didn't confirm, but I suspect that you need build the seed table with something like the following two pieces:
1. https://github.com/dbt-labs/dbt-core/blob/0ef9931d197c996ccd4e3caf641729176eceda99/tests/functional/simple_snapshot/test_basic_snapshot.py#L9-L10
1. https://github.com/dbt-labs/dbt-core/blob/0ef9931d197c996ccd4e3caf641729176eceda99/tests/functional/simple_snapshot/test_basic_snapshot.py#L111-L113

I'm assuming the `seed_newcol` portions in the examples above are not applicable to your case, so you can remove them (but also wouldn't hurt to leave them in initially just to make sure all the CI is working after copy-pasting).

**Comment by RobbertDM** [comment]
Hey @dbeatty10 thanks a lot for the detailed help, I wanted to give up after getting nothing useful from CI.
Where did you get that CI output? 
I looked at this https://github.com/dbt-labs/dbt-core/actions/runs/4127200194/jobs/7130178699 but that doesn't give me your output.

**Comment by dbeatty10** [comment]
Happy to help @RobbertDM !

I saw that error listed [here](https://github.com/dbt-labs/dbt-core/actions/runs/4112159554/jobs/7097304796#step:8:1445).

It sounds like you looked at this top one:

image

And I scrolled down a bit and dug into one of these (which might all be basically identical):

image

**Review by iknox-fa — DISMISSED** [review]
LGTM

**Comment by iknox-fa** [comment]
@RobbertDM 
I made a few small tweaks to your test case for you.  You had a few issues:
* You were trying to combine a class-based approach with a function-based approach, and pytest was having none of it.  
* You weren't finding the seed table because you hadn't actually populated it by running the SQL file
* You need to run `snapshot` to test your fix, not `run`, and crucially, you need to do it twice because the bug only happened when the snapshot table already exists in the DW.
* You had a bunch of unneeded fixtures.  They didn't hurt anything, but they weren't being used.

Should be all set now. :)

(I also added a comment to the macro since we don't want someone coming along and cleaning up that whitespace you added at later date)

@VersusFacit 
Mea Culpa for jumping in on an adapter ticket.  I got tagged for review somehow and got halfway through before I realized I probably shouldn't have been. 🤓

**Review by VersusFacit — APPROVED** [review]

**Comment by VersusFacit** [comment]
Huge thanks to everyone for keeping us alive and getting it across the finish line

**Comment by RobbertDM** [comment]
Woah, thanks a lot for all the help and explanation everyone. I'll definitely come back to this PR if I make another one.
